### PR TITLE
Detector-Competition-Feat: Add Privacy.com API key detector

### PR DESCRIPTION
**Pull request by lc**
### Description:
This pull request adds a detector for https://privacy.com. This service allows individuals and enterprises to create virtual cards to protect their card info. 

The documentation can be found [here](https://privacy-com.readme.io/docs/api-basics)
### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

Tests are passing:
CleanShot 2023-10-12 at 08 32 53

### CODE REVIEW

File: pkg/detectors/privacy/privacy.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`\b([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12})\b`)
```
Comment (zricethezav) [inline pkg/detectors/privacy/privacy.go:25]: @lc same goes for this regex, could we include `privacy` using `RegexPrefix` here?

Comment (lc) [reply pkg/detectors/privacy/privacy.go]: Done

**Comment by zricethezav** [comment]
@lc just gotta fix conflicts then this is good to go

**Comment by lc** [comment]
@zricethezav – done.

**Review by zricethezav — APPROVED** [review]

**Pull request merged**